cmd/scanner-trivy/main.go

```diff
@@ -55,10 +61,21 @@ func run(info etc.BuildInfo) error {
 		return fmt.Errorf("checking config: %w", err)
 	}
 
-	worker := queue.NewWorker(config.JobQueue)
+	var (
+		store persistence.Store
+		worker queue.Worker
+		enqueuer queue.Enqueuer
+	)
+
+	switch strings.ToLower(config.DatabaseType) {
```

Comment:
Looks like a good candidate for Factory Design Pattern:
https://en.wikipedia.org/wiki/Factory_method_pattern

Reply:
To be honest, I thought about it but didn't want to complicate things further since there are only 2 possibilities so far. At the end, maybe I should implement it. Do you have a strong opinion on that?

Reply:
No I don't have a strong opinion on that just wanted to mention it.